### ISSUE TITLE
[apex] False negative for ApexCRUDViolation inside of property getter/setter

### ISSUE THREAD

**Issue by abeaugustijn** [pmd/pmd#4019]
**Affects PMD Version:** 
6.46.0

**Rule:**
ApexCRUDViolation

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_apex_security.html#apexcrudviolation>

**Description:**

**Code Sample demonstrating the issue:**

```java
	private Id hourCreditTypeId {
		get;
		set {
			this.hourCreditTypeId = value;

			Hour_Credit_Type__c creditType = [SELECT Method__c FROM Hour_Credit_Type__c WHERE Id = :value];
			this.creditMethod = creditType.Method__c;
		}
	}

```

**Expected outcome:**

PMD should report a violation at line 6, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*
